### ISSUE TITLE
applying "try this" from clippy::try_err causes "return return"

### ISSUE THREAD

**Issue by ericseppanen** [rust-lang/rust-clippy#7103]
I tried this code:

```rust
pub fn testcase(x: bool) -> Result<(), String> {
    if x {
        return Err("ouch".to_string())?;
    }
    Ok(())
}
```

I expected to see this happen:

Clippy should suggest removing the question mark but leaving everything else the same, i.e.

```rust
pub fn testcase(x: bool) -> Result<(), String> {
    if x {
        return Err("ouch".to_string());
    }
    Ok(())
}
```

Instead, this happened:
```text
warning: returning an `Err(_)` with the `?` operator
 --> src/lib.rs:5:16
  |
5 |         return Err("ouch".to_string())?;
  |                ^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `return Err("ouch".to_string())`
  |
```

Note where the carets point. If I apply this automatically (e.g. in vscode by clicking "try this") then the resulting code is:
```rust
pub fn testcase(x: bool) -> Result<(), String> {
    if x {
        return return Err("ouch".to_string());
    }
    Ok(())
}
```

### Meta

- `cargo clippy -V`: clippy 0.1.51 (2fd73fa 2021-03-23)
- `rustc -Vv`:
  ```
rustc 1.51.0 (2fd73fabe 2021-03-23)
binary: rustc
commit-hash: 2fd73fabe469357a12c2c974c140f67e7cdd76d0
commit-date: 2021-03-23
host: x86_64-unknown-linux-gnu
release: 1.51.0
LLVM version: 11.0.1
```
